wemake_python_styleguide/visitors/ast/imports.py

```diff
@@ -74,39 +62,74 @@ def check_alias(self, node: AnyImport) -> None:
                         VagueImportViolation(node, text=alias.name),
                     )
 
-    def check_same_alias(self, node: AnyImport) -> None:
+    def _check_same_alias(self, node: AnyImport) -> None:
         for alias in node.names:
             if alias.asname == alias.name and not self._options.i_control_code:
                 self._error_callback(
                     SameAliasImportViolation(node, text=alias.name),
                 )
 
-    def check_protected_import(self, node: ast.Import) -> None:
-        self._check_protected_names(
-            chain.from_iterable(
-                [alias.name.split('.') for alias in node.names],
-            ),
-            ProtectedModuleViolation(node),
-        )
 
-    def check_protected_import_from(self, node: ast.ImportFrom) -> None:
-        self._check_protected_names(
-            imports.get_import_parts(node),
-            ProtectedModuleViolation(node),
-        )
-        self._check_protected_names(
-            [alias.name for alias in node.names],
-            ProtectedModuleMemberViolation(node),
+@final
+class _ImportValidator(_BaseImportValidator):
+    """Validator of ``ast.Import`` nodes."""
+
+    def validate(self, node: AnyImport) -> None:
+        node = cast(ast.Import, node)
```

Comment:
We try not use `cast`. It should be a bug in Python typing to actually use it.
Can you please refactor this place once again?

Reply:
Sure, but to do that I need to use everywhere `AnyImport` instead of concrete import type that is validating in given method, is that ok?

e,g `def _check_dotted_raw_import(self, node: AnyImport) -> None:` instead of `def _check_dotted_raw_import(self, node: ast.Import) -> None:`